pkg/machines/libvirt-common.js

```diff
@@ -905,15 +905,20 @@ export function updateDisk(domXml, diskTarget, readonly, shareable) {
     return tmp.innerHTML;
 }
 
-export function unknownConnectionName(action, libvirtServiceName) {
+export function unknownConnectionName(action, libvirtServiceName = undefined) {
     return dispatch => {
         return cockpit.user().done(loggedUser => {
             const promises = Object.getOwnPropertyNames(VMS_CONFIG.Virsh.connections)
                     .filter(
                         // The 'root' user does not have its own qemu:///session just qemu:///system
                         // https://bugzilla.redhat.com/show_bug.cgi?id=1045069
                         connectionName => canLoggedUserConnectSession(connectionName, loggedUser))
-                    .map(connectionName => dispatch(action(connectionName, libvirtServiceName)));
+                    .map(connectionName => {
+                        if (libvirtServiceName)
+                            return dispatch(action(connectionName, libvirtServiceName));
+                        else
+                            return dispatch(action(connectionName));
+                    });
```

Comment:
Please leave the old simpler code  -- passing an extra `undefined` behaves exactly like not passing it at all. In other words, you don't need to modify this function at all, but you should rename it for clarity, or at least add a comment what it does.